File: src/main/java/com/flexpoker/config/ForceHttpsFilter.java
package com.flexpoker.config;

import java.io.IOException;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class ForceHttpsFilter implements Filter {

    public static final String X_FORWARDED_PROTO = "x-forwarded-proto";

    @Override
    public void destroy() {
        // nothing to do
    }

    @Override
    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
            throws IOException, ServletException {
        var request = (HttpServletRequest) servletRequest;
        var response = (HttpServletResponse) servletResponse;

        if (request.getHeader(X_FORWARDED_PROTO) != null //
                && request.getHeader(X_FORWARDED_PROTO).indexOf("https") != 0 //
                && request.getServerName() != "localhost") {
            response.sendRedirect("https://" + request.getServerName()
                    + (request.getServletPath() == null ? "" : request.getServletPath()));
            return;
        }

        filterChain.doFilter(request, response);
    }

    @Override
    public void init(FilterConfig arg0) throws ServletException {
        // nothing to do
    }

}

File: src/main/kotlin/com/flexpoker/config/ForceHttpsFilter.kt
package com.flexpoker.config

import jakarta.servlet.Filter
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import jakarta.servlet.FilterChain
import jakarta.servlet.ServletRequest
import jakarta.servlet.ServletResponse
import jakarta.servlet.FilterConfig

class ForceHttpsFilter : Filter {

    companion object {
        const val X_FORWARDED_PROTO = "x-forwarded-proto"
    }

    override fun destroy() {
        // nothing to do
    }

    override fun doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
        val request = servletRequest as HttpServletRequest
        val response = servletResponse as HttpServletResponse
        val servletPath = if (request.servletPath == null) "" else request.servletPath

        if (request.getHeader(X_FORWARDED_PROTO) != null
            && request.getHeader(X_FORWARDED_PROTO).indexOf("https") != 0
            && request.serverName !== "localhost") {
            response.sendRedirect("https://" + request.serverName + servletPath)
            return
        }
        filterChain.doFilter(request, response)
    }

    override fun init(arg0: FilterConfig) {
        // nothing to do
    }

}